src/cpp/_pythia8.cpp

```diff
@@ -17,9 +18,13 @@ using namespace pybind11::literals;
 float charge_from_pid(const ParticleData &pd, int pid)
 {
     auto pptr = pd.findParticle(pid);
-    assert(pptr); // never nullptr if charge_from_pid is used on particles produced by Pythia
+    // return NaN if unknown pid is met
+    if (pptr == nullptr) {
+        return std::numeric_limits<float>::quiet_NaN();
+    } else {
     // ParticleData returns partice even if anti-particle pid is used
-    return pid == pptr->id() ? pptr->charge() : -pptr->charge();
+        return pid == pptr->id() ? pptr->charge() : -pptr->charge();
+    }
```

Comment:
Remove the else. The code was already correct before and better style. A function which returns something should always have an unconditional return statement at the end for clarity.

Reply:
Done